### ISSUE TITLE
C++17

### ISSUE THREAD

**Issue by kariem2k** [rive-app/rive-sokol#10]
Hello,

First I want to thank you for this project! 

Some places are C++20 only (mainly the designated initializers). Can I make a pull request that fixes this to be C++17? or is it intentional?

Thanks

**Comment by JCash** [comment]
Hi @kariem2k !

I think we already _do_ build with `-std=c++17`? (Since that's what `rive-cpp` uses).

What's the place you say isn't working for you using the build script?

**Comment by kariem2k** [comment]
Thanks @JCash for your reply.

Places like these use C style initializers which (AFAIK) has only relatively limited support C++20+ 

https://github.com/rive-app/rive-sokol/blob/fb34a6cae9fe0a43e7b12587ff2da5cd2a6b0489/src/app.cpp#L353-L357
https://github.com/rive-app/rive-sokol/blob/fb34a6cae9fe0a43e7b12587ff2da5cd2a6b0489/src/rive/rive_render_private.cpp#L132-L135
https://github.com/rive-app/rive-sokol/blob/fb34a6cae9fe0a43e7b12587ff2da5cd2a6b0489/src/app.cpp#L503-L520
Thanks

**Comment by JCash** [comment]
Well, I'm not against it per se, if the compilers don't support it.
Do you have an example of a compiler that doesn't support this? (e.g. we use clang which seems to support it)

**Comment by kariem2k** [comment]
@JCash  yes I am using Visual Studio. I believe that clang enabled this extension by default even if it is not part of the standard.

This is a little example (compiled with C++17 and -pedantic): 

https://godbolt.org/z/Gac9fM5r3

Here is a screenshot from the compilation result

![image](https://user-images.githubusercontent.com/298507/135904385-236e6100-8416-4878-a94b-27a108208ed2.png)

**Comment by JCash** [comment]
Then I say we should move to use C++17 code, since that's what `rive-cpp` uses currently.
